c++/CMakeLists.txt

```diff
@@ -0,0 +1,11 @@
+cmake_minimum_required(VERSION 2.8)
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror")
+set(CMAKE_CXX_FLAGS_DEBUG "-g3")
+
+project(puzzles)
+
+add_executable(../out/binary-tree ../src/binary-tree.cc)
+add_executable(../out/linked-list ../src/linked-list.cc)
+add_executable(../out/vector ../src/vector.cc)
```

Comment:
One thing you can do, which I do, is to do something like:

``` cmake
add_subdirectory(src)
```

And then in `src/` create another `CMakeLists.txt` which has the `add_executable` lines. That way the main/root cmake file is clean and it makes it easier to add more subdirectories in the future.